### PR TITLE
doc: use new feature gate for c_void type

### LINKED ISSUE #55619: `c_void` has wrong rustc support level in docs
The documentation page for [`core::ffi::c_void`](https://doc.rust-lang.org/core/ffi/enum.c_void.html) says it's available at version 1.1.0, however, it's only available at version 1.30.0.

### LINKED ISSUE #63694: core::ffi::c_void is marked as "stable since 1.1", but it was added in 1.30
The title says it all really.

* Missing `ffi` entirely: https://doc.rust-lang.org/1.29.0/core/
* Present in the new `ffi` module: https://doc.rust-lang.org/1.30.0/core/ffi/index.html

### PR DESCRIPTION
**Pull request by tesuji**
Closes #63694, closes #55619

### CODE REVIEW

File: src/libcore/ffi.rs [hunk 1]
```diff
@@ -18,8 +18,13 @@ use crate::ops::{Deref, DerefMut};
 /// stabilized, it is recommended to use a newtype wrapper around an empty
 /// byte array. See the [Nomicon] for details.
 ///
+/// One could use `std::os::raw::c_void` if they want to support old Rust
+/// compiler down to 1.1.0. After Rust 1.30.0, it was re-exported by
+/// this definition. For more infor mation, please read [RFC 2521].
```
Comment (alex) [inline src/libcore/ffi.rs]:
```suggestion
/// this definition. For more information, please read [RFC 2521].
```

Comment (lzutao) [reply src/libcore/ffi.rs]: Thanks. Just found the typo.

**Comment by Centril** [comment]
I know nothing about this :) r? @dtolnay

**Review by dtolnay — APPROVED** [review]
Thanks!

**Pull request merged**